src/sentry/profiles/flamegraph.py

```diff
@@ -87,17 +90,59 @@ def get_span_intervals(
             "organization_id": organization_id,
         },
     )
-    return raw_snql_query(
+    data = raw_snql_query(
         request,
         referrer=Referrer.API_STARFISH_PROFILE_FLAMEGRAPH.value,
     )["data"]
+    spans_interval = []
+    for row in data:
+        start_ns = (int(datetime.fromisoformat(row["start_timestamp"]).timestamp()) * 10**9) + (
+            row["start_ms"] * 10**6
+        )
+        end_ns = (int(datetime.fromisoformat(row["end_timestamp"]).timestamp()) * 10**9) + (
+            row["end_ms"] * 10**6
+        )
+        interval = {}
+        interval["transaction_id"] = row["transaction_id"]
+        interval["start_ns"] = str(start_ns)
+        interval["end_ns"] = str(end_ns)
+        spans_interval.append(interval)
+    return spans_interval
+
+
+def get_span_intervals_from_nodestore(
+    project_id: str,
+    span_group: str,
+    transaction_ids: List[str],
+) -> List[Dict[str, Any]]:
+    spans_interval = []
+    for id in transaction_ids:
+        nodestore_event = eventstore.backend.get_event_by_id(project_id, id)
+        data = nodestore_event.data
+        for span in data.get("spans", []):
+            if span["hash"] == span_group:
+
+                start_sec, start_us = map(int, str(span["start_timestamp"]).split("."))
+                end_sec, end_us = map(int, str(span["timestamp"]).split("."))
+
+                start_ns = (start_sec * 10**9) + (start_us * 10**3)
+                end_ns = (end_sec * 10**9) + (end_us * 10**3)
```

Comment:
timestamp is a float, if I multiply it by `10**9` I get a result with the number in scientific notation. Since I need to pass it as a string to I have to first cast to int and then str. In doing so, it loses some precision though.

That's why I'm not doing `timestamp * 1e9` directly.